### ISSUE TITLE
northdirection + crop results in wrong coordinates and compass in output

### ISSUE THREAD

**Issue by CounterPillow** [overviewer/Minecraft-Overviewer#1144]
As reported by two people on IRC, cropping and changing the northdirection results in wrong (flipped) coordinates for the world, and also a wrong compass.

Config one:

```python
worlds["world"] = "/home/DOKDOR/1.7.9/world"
 
renders["normalrender"] = {
    "world": "world",
    "title": "PvE daytime",
    "rendermode": "normal",
    "dimension": "overworld",
    "crop": (-2000, -2000, 2000, 2000),
    "northdirection":"upper-left",
    "imgformat":"jpeg",
    "imgquality": 70
}
 
renders["reversednormalrender"] = {
    "world": "world",
    "title": "PvE daytime Reversed",
    "rendermode": "normal",
    "dimension": "overworld",
    "crop": (-2000, -2000, 2000, 2000),
    "northdirection":"lower-right",
    "imgformat":"jpeg",
    "imgquality": 70
}
 
outputdir = "/var/www/html"
```

Config 2:

```python
from optimizeimages import optipng

worlds["world"] = "/srv/minecraft/world"
processes = 4
northdirection = "upper-right"

renders["Day"] = {
    "world": "world",
    "title": "Day",
    "rendermode": smooth_lighting,
    "dimension": "overworld",
    "crop": (-50000, -50000, 50000, 50000),
    "optimizeimg": [optipng()]
}

outputdir = "/my/obfuscated/outputdir/"
showspawn = False
```

Both cases were reported after the recent changes of the cropping option merged in, so perhaps it is related.

**Comment by tdelcourt** [comment]
Ive noticed the same thing.

```python
renders["walie8-crop-smooth_lighting-ul"] = {
    "world": "walie8",
    "title": "walie8-smooth_lighting-ul",
    "rendermode": smooth_lighting,
    "dimension": "overworld",
    "northdirection": "upper-left",
    "crop": (0, 0, 100, 100),
}
renders["walie8-crop-smooth_lighting-ur"] = {
    "world": "walie8",
    "title": "walie8-smooth_lighting-ur",
    "rendermode": smooth_lighting,
    "dimension": "overworld",
    "northdirection": "upper-right",
    "crop": (0, 0, 100, 100),
}
```
Gives me two different spawn points (248, 211) and (261, 224) and two different and non-overlapping pieces of land (albeit verry close to each other)

Also, without cropping, the 0,0 blocks fo the two full-map renderings are actually something like 15 blocks apart on the x axis. 

And when I tried to render a cropped piece of land (500x500 blocks) far from the spawn point, it was rendered in upper-left, but not at all in the three other orientations. But this perticular piece of land is minimum 100 blocks away from the edge of the map, so the error is verry big.

**Comment by narn-l** [comment]
I have also run into this issue. Using a crop, a lower-right north direction causes all coordinates to be inverted (500, 1000 on the world becomes -500,-1000 indicated on the overviewer). This includes the spawn point and so also the map centre, so the overviewer covers an entirely wrong section of the world.

**Comment by thwarted** [comment]
I'm running into this too.  If I play with the crop coordinates to get it to render the area I want for `northdirection` values other than `upper-left`, choosing a selected render also doesn't move the view to the new location (I suppose this is actually good, otherwise this problem wouldn't be so evident).  I have a crop of `(78000,1300,81000,-1000)`, and there isn't much else in my world, so unless I play with the crop offsets to reposition it, the other ones come up blank.

**Comment by Socolin** [comment]
Seem to come from https://github.com/overviewer/Minecraft-Overviewer/commit/b6ac54a2b61d0b8f45adab36ac5179e83bcef710

To be exact, here: https://github.com/overviewer/Minecraft-Overviewer/commit/b6ac54a2b61d0b8f45adab36ac5179e83bcef710#diff-762d110af28fb64b4e311e28c69c3787L491

The order of Crop and Rotate change so: 

For the POI the problem is here: https://github.com/overviewer/Minecraft-Overviewer/blob/master/overviewer_core/tileset.py#L594
The crop region is applied at the end, so the property `north_dir` is not found, it except and the value is set to `0` in `overviewerConfig.js`

For the stange render, I still have to look where is the probleme, but it may due to the inversion of Crop and Rotate

I'm doing some test and post a patch if I found a way to fix it

**Comment by agrif** [comment]
This ought to be fixed now that #1282 is merged. If you still have it, please let us know!
